x/launch/types/initial_genesis.go

```diff
@@ -1,4 +1,28 @@
 package types
 
+import "errors"
+
 // InitialGenesis defines the interface for initial genesis types
-type InitialGenesis interface{}
+type InitialGenesis interface {
+	Validate() error
+}
+
+var _ InitialGenesis = &DefaultInitialGenesis{}
+
+// Validate implements InitialGenesis
+func (DefaultInitialGenesis) Validate() error { return nil }
+
+var _ InitialGenesis = &GenesisURL{}
+
+const HashLength = 64
+
+// Validate implements InitialGenesis
+func (g GenesisURL) Validate() error {
+	if g.Url == "" {
+		return errors.New("no url provided")
+	}
+	if len(g.Hash) != HashLength {
+		return errors.New("hash must be sha256")
+	}
+	return nil
+}
```

Comment:
Should we create separate tests for it? What do you think?

Reply:
Yep good point. We should maybe think about having a test coverage CI step

Reply:
we can use the [codecov](https://about.codecov.io/language/go/)